### DISCUSSION TITLE
Using bool types?

### DISCUSSION THREAD

**Discussion by winterz** [libical/libical#842] [General]
Now that we require C99 in the master branch...

any thoughts about using the bool return type (as in <stdbool.h>) where appropriate?
In icaltime alone we should be returning booleans for these:
```
LIBICAL_ICAL_EXPORT int icaltime_is_null_time(const struct icaltimetype t);
LIBICAL_ICAL_EXPORT int icaltime_is_valid_time(const struct icaltimetype t);
LIBICAL_ICAL_EXPORT int icaltime_is_date(const struct icaltimetype t);
LIBICAL_ICAL_EXPORT int icaltime_is_utc(const struct icaltimetype t);
LIBICAL_ICAL_EXPORT int icaltime_is_leap_year(const int year);
LIBICAL_ICAL_EXPORT int icaltime_span_overlaps(icaltime_span *s1, icaltime_span *s2);
LIBICAL_ICAL_EXPORT int icaltime_span_contains(icaltime_span *s, icaltime_span *container);
```
and there are others 'int' functions scattered about that should be boolean
there may be functions with 'int' args that should be using bools too, but I haven't looked for those yet.

Good idea? Bad idea?

**Comment by ksmurchison** [comment]
Works for me
